smartsim/_core/control/controller.py

```diff
@@ -512,6 +512,14 @@ def _launch_step(
         :raises SmartSimError: if launch fails
         """
         try:
+            if (
+                entity.name in self._jobs.completed
+                or self._jobs.jobs
+                or self._jobs.db_jobs
```

Comment:
I'm guessing this is not going to do what you intended: This clause will always eval `True` if `self._jobs.jobs` or `self._jobs.db_jobs` is a non-empty `dict[str, Job]`.

Maybe
```suggestion
                entity.name in self._jobs.completed
                or entity.name in self._jobs.jobs
                or entity.name in self._jobs.db_jobs
```
?